### PR TITLE
Add argument to `draw3p` for checking that custom iframe is embedded into allowed origin.

### PR DESCRIPTION
**Pull request by cramforce**

### CODE REVIEW

File: 3p/integration.js [hunk 1]
```diff
@@ -338,6 +345,42 @@ export function validateAllowedTypes(window, type, allowedTypes) {
 }
 
 /**
+ * Check that parent host name was whitelisted.
+ * @param {!Window} window
+ * @param {!Array<string>} allowedHostnames Suffixes of allowed host names.
```
Comment (erwinmombay) [inline 3p/integration.js]: add `@return`

File: 3p/integration.js [hunk 2]
```diff
@@ -153,8 +154,11 @@ function masterSelection(type) {
  *        no arguments. Configuration is expected to be modified in-place.
  * @param {!Array<string>=} opt_allowed3pTypes List of advertising network
  *     types you expect.
+ * @param {!Array<string>=} opt_allowedEmbedddingOrigins List of domain suffixes
```
Comment (erwinmombay) [inline 3p/integration.js]: Embed(d)ding -> Embedding?

File: 3p/integration.js [hunk 3]
```diff
@@ -338,6 +345,42 @@ export function validateAllowedTypes(window, type, allowedTypes) {
...
+  const onDefault = hostname == 'cdn.ampproject.org';
+  if (onDefault) {
+    // If we are on the cache domain, parse the source hostname from
+    // the referrer. The referrer is used because it should be
+    // trustable.
+    hostname = parseUrl(getSourceUrl(window.document.referrer)).hostname;
+  }
+  for (let i = 0; i < allowedHostnames.length; i++) {
+    // Either the hostname is exactly as whitelisted…
+    if (allowedHostnames[0] == hostname) {
```
Comment (erwinmombay) [inline 3p/integration.js]: should this be `allowedHostnames[i]` ?

Comment (cramforce) [inline 3p/integration.js]: Nice catch. Added test

File: 3p/integration.js (hunk 2 above)
Comment (cramforce) [inline 3p/integration.js]: Done

File: 3p/integration.js (hunk 1 above)
Comment (cramforce) [inline 3p/integration.js]: Removed return values.

**Comment by cramforce** [comment]
PTAL

**Comment by erwinmombay** [comment]
@cramforce LGTM

**Pull request merged**